src/compute-client/src/types/sinks.rs

```diff
@@ -29,7 +29,7 @@ pub struct ComputeSinkDesc<S: 'static = (), T = mz_repr::Timestamp> {
     pub from: GlobalId,
     pub from_desc: RelationDesc,
     pub connection: ComputeSinkConnection<S>,
-    pub as_of: SinkAsOf<T>,
+    pub with_snapshot: bool,
     pub up_to: Antichain<T>,
```

Comment:
We could make a similar change with the sink `up_to`, replacing it with the dataflow `until` frontier. But I'm not sure what that entails.